### DISCUSSION TITLE
Linux PC build for meep

### DISCUSSION THREAD

**Discussion by dalarev** [NanoComp/meep#2925] [Q&A]
Hello, I am looking to build a workhorse linux PC optimized for heavy meep use and would appreciate guidance/tips from those who know computer architecture, MPI, and meep better than I do. The goal is to build a suitable system for large 3D simulations.

Below are my initial estimates for RAM, CPU and GPU resources based on meep documentation and others. I would very much appreciate any guidance the community can offer, as I am by no means an expert on this.

--

**RAM: 256 GB minimum, expandable to 512 and beyond.**

Following the [memory requirements estimate](https://meep.readthedocs.io/en/latest/FAQ/#what-is-an-estimate-of-the-memory-requirements-of-a-simulation-given-only-its-cell-size-and-resolution) guidance, initial calculations of an expected use-case give a lower bound of 158 GB. Since this does not account for PMLs, materials, and others, I figure opting for a minimum of 256 GB of RAM, with the option to expand to 512 or beyond, is a reasonable guess.

**CPU: 96 cores on a single cpu, expandable to 2x CPUs**

Currently looking at the [AMD EPYC 9654](https://www.amd.com/en/products/processors/server/epyc/4th-generation-9004-and-8004-series/amd-epyc-9654.html). The reason for this is based on my reading of the [expected speed up from parallel meep](https://meep.readthedocs.io/en/latest/FAQ/#should-i-expect-linear-speedup-from-the-parallel-meep). My take from this is that it's better to have more cores in a single node to avoid network communication delays and cache contention. But I am far from an expert on this and could be totally incorrect. I assume, but do not know for sure, that AMD CPUs support some form of MPI that's compatible with meep.

**GPU: Whatever is the top of the line, commercial option up to $2k**

Meep doesn't support GPU acceleration, so really a good GPU is only needed to render potentially large 3D models and results post-processing.

--

Thanks again for the help! I'm happy to add any clarity needed, as there is no one "best" build and is case-specific.

**Comment by sheevy** [comment]
Those AMD with 3D V-Cache are quite well suited for meep. For ram, you're using a server motherboards, I don't think you can choose speed. Make sure you populate all RAM slots and don't leave any unused. IIRC, those CPUs are mounted on 2 socket MB with 24 slots. So I think you should do 384 or 768. Also, I'm not sure you actually can just buy it with a single CPU, you'd need to check with your vendor.
